File: LLM-Translations/GPT-4/avatar/Python/Go/codeforces_356_A.go

package main

import (
	"fmt"
	"bufio"
	"os"
	"strings"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	parts := strings.Split(scanner.Text(), " ")
	n, _ := strconv.Atoi(parts[0])
	m, _ := strconv.Atoi(parts[1])
	out := make([]int, n)
	comp := make([]int, n)
	for i := range comp {
		comp[i] = i + 2
	}
	for i := 0; i < m; i++ {
		scanner.Scan()
		parts = strings.Split(scanner.Text(), " ")
		l, _ := strconv.Atoi(parts[0])
		r, _ := strconv.Atoi(parts[1])
		x, _ := strconv.Atoi(parts[2])
		t := l
		for t <= r {
			nextVal := comp[t-1]
			if out[t-1] == 0 && t != x {
				out[t-1] = x
			}
			if t >= x {
				comp[t-1] = r + 1
			} else {
				comp[t-1] = x
			}
			t = nextVal
		}
	}
	fmt.Println(out)
} // End of Code

File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_356_A.c
#include<stdio.h>

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    int out[n], comp[n];
    for(int i = 0; i < n; i++) {
        out[i] = 0;
        comp[i] = i + 2;
    }
    for(int i = 0; i < m; i++) {
        int l, r, x;
        scanf("%d %d %d", &l, &r, &x);
        int t = l;
        while(t <= r) {
            int next_val = comp[t - 1];
            if(out[t - 1] == 0 && t != x) {
                out[t - 1] = x;
            }
            comp[t - 1] = (t >= x) ? r + 1 : x;
            t = next_val;
        }
    }
    for(int i = 0; i < n; i++) {
        printf("%d ", out[i]);
    }
    return 0;
}

// End of Code
